webapp/src/auth.ts

```diff
@@ -327,4 +371,19 @@ function clearState() {
     data.invalidate(NEEDS_SETUP);
 }
 
+function userPreferencesHandler(path: string): UserPreferences {
+    if (!state_.preferences) {
+        // TODO @darzu: what's the right way to handle defaults here?
+        state_.preferences = {
+            highContrast: false,
+            language: "dz2",
+        }
+        fetchUserPreferencesAsync();
+    }
+    return state_?.preferences 
+}
+
 data.mountVirtualApi("auth", { getSync: authApiHandler });
+
+console.log("mounting user-pref") // TODO @darzu: 
+data.mountVirtualApi("user-pref", { getSync: userPreferencesHandler });
```

Comment:
Question - if it's getting all the preferences at once here, will it end up making any change in a single preference rerender all components that refer to any preference (e.g. changing language also rerenders all high contrast elements)? Would it make sense to subscribe to each pref separately ('user-pref:high-contrast', 'user-pref:language') so that each component only listens for the preferences it depends upon?

Reply:
You've got it exactly right. For simplicity I was using user-pref everywhere but we should use :high-contrast etc so we can have more fine-grain subscriptions.

Reply:
Okay, fine-grain subscriptions should be working in theory; gotta do a little testing to make sure